map_view/display-layer-view-state/src/main/java/com/esri/samples/display_layer_view_state/DisplayLayerViewStateSample.java

```diff
@@ -16,132 +16,166 @@
+        }
+      });
 
-      // create a view and set ArcGISMap to it
-      mapView = new MapView();
-      mapView.setMap(map);
+      // create a button to toggle the visibility of the feature layer
+      toggleLayerButton = new Button("Load Layer");
+      toggleLayerButton.setMaxWidth(130);
+
+      // create a listener for clicks on the load layer button
+      toggleLayerButton.setOnAction(event -> {
+
+        // if the feature layer already exists toggle it's visibility to visible
+        if (featureLayer != null) {
+          featureLayer.setVisible(!featureLayer.isVisible());
+        } else {
+          // if the feature layer doesn't exist, create a new feature layer from a portal item
+          final PortalItem portalItem = new PortalItem(new Portal("https://runtime.maps.arcgis.com/"),
+            "b8f4033069f141729ffb298b7418b653");
+          featureLayer = new FeatureLayer(portalItem, 0);
+          //  set a minimum and a maximum scale for the visibility of the feature layer
+          featureLayer.setMinScale(40000000);
+          featureLayer.setMaxScale(4000000);
+          // add the feature layer to the map
+          map.getOperationalLayers().add(featureLayer);
+        }
 
-      // set viewpoint
-      mapView.setViewpoint(new Viewpoint(new Point(-11e6, 45e5, SpatialReferences.getWebMercator()), MIN_SCALE));
-
-      // fires every time a layers' view status has changed
-      mapView.addLayerViewStateChangedListener(e -> {
-        // holds the label that needs to be changed
-        Layer layer = e.getLayer();
-
-        String viewStatus = e.getLayerViewStatus().iterator().next().toString();
-        final int layerIndex = map.getOperationalLayers().indexOf(layer);
-
-        // finding and updating label that needs to be changed
-        switch (layerIndex) {
-          case TILED_LAYER:
-            worldTimeZonesLabel.setText("World Time Zones: " + viewStatus);
-            break;
-          case IMAGE_LAYER:
-            censusLabel.setText("Census: " + viewStatus);
-            break;
-          case FEATURE_LAYER:
-            facilitiesLabel.setText("Facilities: " + viewStatus);
+        // toggle the text on the button
+        if (toggleLayerButton.getText().equals("Load Layer")){
+          toggleLayerButton.setText("Hide Layer");
+        } else {
+          toggleLayerButton.setText("Load Layer");
         }
       });
 
-      // add the map view and control panel to stack pane
+      // create a control panel and add the label and button
+      VBox controlsVBox = new VBox(6);
+      controlsVBox.setAlignment(Pos.TOP_CENTER);
+      controlsVBox.setBackground(new Background(new BackgroundFill(Paint.valueOf("rgba(0,0,0,0.5)"), CornerRadii.EMPTY,
+        Insets.EMPTY)));
+      controlsVBox.setPadding(new Insets(10.0));
+      controlsVBox.setMaxSize(400, 120);
+      controlsVBox.getStyleClass().add("panel-region");
+      controlsVBox.getChildren().addAll(layerViewStatusLabel, toggleLayerButton);
```

Comment:
A shame none of these work properly with the sample viewer. Which is quite bizarre since they work fine with some other samples!

Reply:
Agreed on this point, it was very frustrating to have to pull them out of the css. Something for us to look at.